localstack/services/lambda_/provider.py

```diff
@@ -308,7 +313,11 @@ def on_after_init(self):
         self.router.register_routes()
         get_runtime_executor().validate_environment()
 
+    def on_before_start(self):
+        self.custom_router_rules = ROUTER.add(LambdaCustomEndpoints())
```

Comment:
Since this endpoint does not need a full service loading, this could - instead of adding these endpoints when loading the service - be done in an init hook (similar to `localstack.services.lambda.plugins.validate_configuration`).
This would help you avoid the data plane service router changes, because there's no need for a service detection.

Reply:
Awesome 💯  That's what I was looking for. Thanks for the suggestion @alexrashed 🙏 
Migrated to hooks now 👍